### PR TITLE
Add ability to customize what are extractable files

### PR DESCRIPTION
**Pull request by n1ru4l**
Implements #10

- Moves logic for checking whether a value is a file to the `isFileValue` function.
- Exposes `isFileValue` function
- Adds a new option `isFileValue` to `extractFiles` function (default value is the exported `isfileValue` function)
- Adds tests for ensuring the `isFileValue` option is working correctly

@jaydenseric Could you please provide some feedback? Especially regarding the jsdoc stuff. Is adding a new typedef for `IsFileValueFunction` correct?

### CODE REVIEW

File: src/index.mjs [hunk 1]
```diff
@@ -29,6 +30,18 @@ export { ReactNativeFile } from './ReactNativeFile'
  * ```
  */
 
+/**
+ * @kind typedef
```
Comment (n1ru4l) [inline src/index.mjs]: @jaydenseric Is this the correct way for declaring the type?

File: readme.md [hunk 2]
```diff
@@ -146,13 +172,13 @@ _Object path is property `a`, array index `0`, object property `b`._
 
 A React Native [`File`](https://developer.mozilla.org/docs/web/api/file) substitute for when using [`FormData`](https://developer.mozilla.org/docs/web/api/formdata).
 
-**Type:** [Object](https://mdn.io/object)
```
Comment (n1ru4l) [inline readme.md]: @jaydenseric Any idea why the links got removed?

File: src/index.mjs (hunk 1 above)
Comment (jaydenseric) [reply src/index.mjs]: I added a bit of polish to the JSDoc :) This one needed a description.

File: readme.md (hunk 2 above)
Comment (jaydenseric) [reply readme.md]: Yes, the updated [`jsdoc-md`](https://www.npmjs.com/package/jsdoc-md) dev dependency no longer auto-links these types.

**Comment by koresar** [comment]
That's a useful PR. It would decouple the module from the ReactNative. Nice.

**Comment by n1ru4l** [comment]
> That's a useful PR. It would decouple the module from the ReactNative. Nice.

My intended use-case was in Node.js environments.


@jaydenseric Sorry for bothering you, but this change is floating around for a while now. 😅Is there anything blocking this PR from getting merged?

**Comment by jaydenseric** [comment]
@n1ru4l sorry for the delay, I have just returned from a holiday and am recovering from a cold but I hope to dig into it this week :)

**Comment by andreisocaciu** [comment]
Any news?

**Comment by lynxtaa** [comment]
@jaydenseric sorry to bother you. When this PR will be merged? Do you need any help?

**Comment by jaydenseric** [comment]
@n1ru4l thanks for kicking this off 🙌

**Pull request merged**